File: CPE/Exam/240521/pD_UVA10858.py
"""
    Backtracking

    AC: UVA, CPE, ZeroJudge
    ZeroJudge 上 Python 會 TLE，開輸出入優化也沒用
"""
ans = []
path = []
def dfs(x):
    if x == 1 and len(path) > 1:
        ans.append(path[:])
        return
    pre = path[-1] if path else 2 # 前一個數字
    for y in range(pre, x+1):
        if x % y == 0:
            path.append(y)
            dfs(x//y)
            path.pop()

while True:
    n = int(input())
    if n == 0:
        break
    ans.clear()
    dfs(n)
    print(len(ans))
    for a in ans:
        print(" ".join(map(str, a)))

File: CPE/Exam/240521/pD_UVA10858.cpp
#include <bits/stdc++.h>
using namespace std;
#define endl '\n'

vector<vector<int>> ans;
vector<int> path;

void dfs(int x) {
    if (x == 1 && path.size() > 1) {
        ans.push_back(path);
        return;
    }
    int pre = path.empty() ? 2 : path.back();
    for (int y = pre; y <= x; y++) {
        if (x % y == 0) {
            path.push_back(y);
            dfs(x / y);
            path.pop_back();
        }
    }
}

int main() {
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    int n;
    while (cin >> n && n) {
        ans.clear();
        dfs(n);
        cout << ans.size() << endl;
        for (auto a : ans)
            for (int i = 0; i < a.size(); i++)
                cout << a[i] << (i == a.size() - 1 ? endl : ' ');
    }
    return 0;
}